Merge latest changes from dev into master

richardwu opened the pull request:
We need to update master with dev now that dev is stable. Changes in this merge:

- Fixed numerous exception handling bugs
- Normalization of clean and training data is now consistent 
- Added Travis CI
- Added a unit/integration test
- Fixed a couple of typos in README
- Added `create_db_ubuntu.sh` and `create_pd_user_ubuntu.sh` scripts for initializing postgres
- Added numerous docstrings and comments to functions
- Introduced a posterior probability estimator to approximate the conditional distribution of values for weak labelling and domain generation. This improves weak supervised learning. See parameters `weak_label_thresh`, `domain_prune_thresh`, and `max_domain` (#43)
- Outputting weights for featurizers when `print_fw=True`
- Disable pool creation when `threads <= 1`
- Fixed UTF-8 encoding issue when ingesting training data
- Fixed bug where non-lowercase attribute names caused Postgres to fail
- Added two new test datasets `met` and `Adult`

richardwu committed:
Groundwork for fusion implementation and assorted bug fixes.

Some notable changes:
- Support for loading fusion and/or repair raw datasets.
- Imported flights dataset for fusion testing.
- Restructured test cases to use unittests.
- Removed unnecessary try excepts that caused exceptions to be silenced.
- Fixed a few bugs as a result of silenced exceptions.
- Keep attributes/columns in their original case (instead of lowercasing
everything): modify Postgres queries to quote references to columns.

richardwu committed:
Addressed PR comments re: previous commit.

Commit:
add exception handling in evaluation; return eval stat in the form of list to support batch running

Commit:
normalize clean dataset to match input dataset normalization

Commit:
save featurizer weights

Commit:
occurence

Commit:
occur attr feat

Commit:
add occur attr feat to init

Commit:
rm batch

ScarletGuo committed:
Delete hc.py

Commit:
batch scripts

Commit:
batch scripts

Commit:
relocate create db

Commit:
reloacate hc

Commit:
typo fixed

Commit:
fix relative references in run scripts

Commit:
fix type err

ScarletGuo committed:
Merge pull request #1 from ScarletGuo/zhihan/debugging_mode

fix type err

Commit:
return weights for auto recording in batch runs

ScarletGuo committed:
Merge pull request #2 from ScarletGuo/zhihan/debugging_mode

return weights for auto recording in batch runs

Commit:
adult analysis

ScarletGuo committed:
Update occurfeat.py

Commit:
food

Commit:
ignore

Commit:
fixing bad char in col name

ScarletGuo committed:
fix indentation

ScarletGuo committed:
Update featurize.py

ScarletGuo committed:
get tensor size

ScarletGuo committed:
move print_fw condition checking to holoclean.py

ScarletGuo committed:
move print_fw condition checking to holoclean

ScarletGuo committed:
save featurizer_weights to status and return

ScarletGuo committed:
Merge pull request #5 from HoloClean/dev

use dev as base

ScarletGuo committed:
get tensor size (subclass of tuple)

ScarletGuo committed:
set -1 for na

ScarletGuo committed:
unify flag name and dest

ScarletGuo committed:
change exception handling

ScarletGuo committed:
remove redundant els

ScarletGuo committed:
unmute evaluation error

Commit:
with single vid debugging

Commit:
single vid

Commit:
format

Commit:
format

richardwu committed:
Addressed PR comments: created new examples directory.

Commit:
single

minafarid committed:
Merge pull request #29 from HoloClean/master

Merge pull request #28 from HoloClean/dev

richardwu committed:
Fixed f_name to basename fpath and missing logging imports.

richardwu committed:
Updated README.

minafarid committed:
Merge pull request #18 from richardwu/fusion-impl

Groundwork for fusion implementation and assorted bug fixes

ScarletGuo committed:
Merge branch 'dev' into zhihan/debugging_mode

ScarletGuo committed:
added error logging for "Cannot find attr"

ScarletGuo committed:
do not downcase

ScarletGuo committed:
revert wrong changes

ScarletGuo committed:
Raise preserving traceback

Commit:
Merge pull request #22 from ScarletGuo/zhihan/debugging_mode

Zhihan/debugging mode

richardwu committed:
Fixed encoding issue where dataframes were not encoded as unicode
strings but lookup with Postgres results were in unicode.

richardwu committed:
Updated comment for get_statistics.

richardwu committed:
Added TODO to topk and removed pdb trace.

Commit:
added support for non-learnable featurizer weights

Commit:
fixed bias_flag and new line

minafarid committed:
Minor syntax fix

minafarid committed:
Merge pull request #33 from HoloClean/non-learnable-featurizers

Non learnable featurizers

minafarid committed:
Merge pull request #31 from richardwu/fix_encoding

Fixed encoding issue where dataframes were not encoded as unicode

ScarletGuo committed:
Merge branch 'zhihan/profiler_mode' into dev

laferrieren committed:
adding support for using env varibles

laferrieren committed:
fixing install instructions, liniting as passing through code

minafarid committed:
Merge pull request #35 from laferrieren/fluffy-test

Updating docs + Adding support for env vars

laferrieren committed:
fixing reporting function

richardwu committed:
Remove jupyter checkpoints.

minafarid committed:
Merge pull request #37 from richardwu/dev

Remove jupyter checkpoints

richardwu committed:
Fixed bad reference to status.

minafarid committed:
Merge pull request #36 from laferrieren/repair_bug

fixing reporting function

richardwu committed:
Fixed NullDetector: cannot use isnull() since we convert nulls to
'_nan_'

Commit:
updating domain generation. Still running tests. Evaluation has to be fixed

Commit:
removed naive bayes denominator

minafarid committed:
Add travis-ci integration

Commit:
Merge pull request #38 from HoloClean/mina/travis-ci

Add Travis CI

ScarletGuo committed:
remove local debugging codes

ScarletGuo committed:
remove local debugging code

ScarletGuo committed:
remove local debugging code

ScarletGuo committed:
remove local debugging info

Commit:
revert merges from zhihan/profiler_mode

richardwu committed:
Preliminary work with logistic regression model.

richardwu committed:
added todo comments.

richardwu committed:
Replaced naive bayes with logistic regression model for weak labelling.

richardwu committed:
Fixed updated domain dataframe.

richardwu committed:
Added fast path for 1 featurizer.

richardwu committed:
Fixed CellStatus enum values.

richardwu committed:
Prune domain based on posterior probabilities computed.

richardwu committed:
Fixed Naive Bayes.

richardwu committed:
Fixed repairing recall.

richardwu committed:
Feed one big batch tensor when making predictions for domain pruning.

richardwu committed:
Fixed NaiveBayes.

richardwu committed:
Removed empty lines.

richardwu committed:
Fixed null detector.

richardwu committed:
Do not add all NULL columns as DK.

Commit:
Merge pull request #39 from richardwu/fix_null_detector

Fix NullDetector and regression with status (dev branch)

richardwu committed:
Updated example script and do not weak label for single value cells.

richardwu committed:
Reduce threads to 4 in test.

richardwu committed:
Changed example threads to 4.

richardwu committed:
Added feature names to weight output.

richardwu committed:
Fix call to orig_cnf.

richardwu committed:
Added value column to inf_values_idx table and fixed constraints for
constants.

richardwu committed:
Added adult and met datasets: added cooccurattr featurizer to logistic
regression. Only allocate pool if threads > 1.

richardwu committed:
Revert back to hospital dataset in example.

richardwu committed:
Removed OccurFeaturizer from examples. Deleted temporary dataframes to
alleviate memory issues.

richardwu committed:
Added smaller hospital test example.

minafarid committed:
Merge pull request #46 from richardwu/feature_names

Added feature names to weight output

richardwu committed:
Merge pull request #43 from richardwu/fix_domain_and_weak_labels

Use Logistic Regression with co-occur to generate weak labels

minafarid commented on dataset/dataset.py:
```diff
@@ -37,6 +19,11 @@ class AuxTables(Enum):
     inf_values_idx = 6
     inf_values_dom = 7
 
+class CellStatus(Enum):
```
Can we please add some documentation about this Enum and 1 line about what each value means

minafarid commented on dataset/dataset.py:
```diff
@@ -153,32 +174,43 @@ def generate_aux_table_sql(self, aux_table, query, index_attrs=False):
+            2. self.single_attr_stats ({ attribute -> { value -> count } })
+              the frequency (# of entities) of a given attribute-value
+            3. self.pair_attr_stats ({ attr1 -> { attr2 -> {val1 -> {val2 -> count } } } })
+              where DataFrame contains 3 columns:
```
I think this is no longer a dataframe

minafarid commented on dataset/dataset.py:
```diff
@@ -188,10 +220,10 @@ def get_statistics(self):
-          2. self.pair_attr_stats ({ attr1 -> { attr2 -> DataFrame } } where
-            DataFrame contains 3 columns:
+          2. self.pair_attr_stats ({ attr1 -> { attr2 -> {val1 -> {val2 -> count } } } })
+            where DataFrame contains 3 columns:
```
Again, I think that the returned stats are not in a dataframe but rather the dictionary that is described in the previous line.

minafarid commented on domain/domain.py:
```diff
@@ -97,44 +97,44 @@ def setup_attributes(self):
+                  <val2>: all values for attr2 that appeared at least once with <val1>
+                  <count>: frequency (# of entities) where attr1: <val1> AND attr2: <val2>
 
         to a flattened 4-level dictionary { attr1 -> { attr2 -> { val1 -> [Top K list of val2] } } }
```
No dataframes, and this no longer flattens, it just filters values above `tau`

jonmio committed:
Fix broken link in README for Conda installation

minafarid committed:
Merge pull request #48 from jonmio/patch-1

Fix broken link in README for Conda installation

minafarid committed:
Py3: upgrade torch version

minafarid committed:
Py3: travis python version

minafarid commented on domain/domain.py:
```diff
@@ -57,16 +59,14 @@ def find_correlations(self):
         # Computer correlation across attributes
-        m_corr = d.corr()
+        m_corr = df.corr()
         self.correlations = m_corr
```
merge the previous two lines?

minafarid commented on domain/domain.py:
```diff
@@ -259,14 +339,13 @@ def get_domain_cell(self, attr, row):
-                        # error since co-occurrence must be at least 1 (since the
-                        # current row counts as one co-occurrence).
-                        raise Exception(missing_val)
+                    if not pd.isnull(row[attr]):
```
can we revert the logic to `continue` if `pd.isnull` to make it more readable

minafarid committed:
Py3: upgrade psycopg2 version

minafarid committed:
Merge branch 'dev-py3' of github.com:HoloClean/holoclean into dev-py3

minafarid committed:
Documentation and styling [IP]

minafarid committed:
Removed deprecated OccurFeaturizer

minafarid committed:
Styling

minafarid committed:
Fixed tests

minafarid committed:
Addressed PR Comments

minafarid committed:
Merge pull request #49 from HoloClean/dev-pr47

Documentation and styling

minafarid committed:
Travis python  2.7 and 3.6

minafarid committed:
Travis python 3.7

minafarid committed:
Readme to support python 3

minafarid committed:
Readme minor

minafarid committed:
Readme styling

minafarid committed:
query format

minafarid committed:
Remove python 3.7 from travis

minafarid committed:
SQL syntax

minafarid committed:
Revert SQL syntax

minafarid committed:
Fix sql query

richardwu committed:
Memoize get_corr_attributes and a few style changes.

Memoizing get_corr_attributes reduces runtime by about ~20%.

minafarid committed:
virtualenv readme

minafarid committed:
Merge pull request #50 from HoloClean/dev-py3

Support Python3

oattia committed:
some styling, adding comments, fixing some wrong documentatin references

richardwu committed:
Removed OccurFeaturizer in RecurrentLogistic.

minafarid committed:
Merge branch 'dev' into styling

minafarid committed:
Merge pull request #51 from richardwu/memoize_corr

Performance improvements for LR by memoizing get_corr_attributes and a few style changes.

minafarid committed:
Merge pull request #52 from oy-attia/styling

some styling, adding comments, fixing some wrong documentatin references

jonmio commented on README.md:
````diff
@@ -14,125 +19,147 @@ predictions, and insights form noisy, incomplete, and erroneous data.
 conda environment by running
 ```bash
-$ source activate holo_env
+$ source activate hc36
````
Consider adding instructions to deactivate venv when finished with `source deactivate`

minafarid replied:
Done

jonmio commented on dataset/dataset.py:
```diff
@@ -65,44 +59,65 @@ def __init__(self, name, env):
 
+        :param name: (str) name to initialize dataset with.
+        :param fpath: (str) filepath to CSV file.
+        :param na_values: (str) value that identifies a NULL value
```
```suggestion
        :param na_values: (List[str] or str) value that identifies a NULL value.
```

jonmio commented on dataset/dataset.py:
```diff
@@ -65,44 +59,65 @@ def __init__(self, name, env):
             # Call to store to database
             self.raw_data.store_to_db(self.engine.engine)
-            status = 'DONE Loading '+f_name
+            status = 'DONE Loading {fname}'.format(fname=os.path.basename(fpath))
```
Nit: Seems like both str.format() and % are used to format strings in the codebase

jonmio commented on dataset/dataset.py:
```diff
@@ -153,56 +172,72 @@ def generate_aux_table_sql(self, aux_table, query, index_attrs=False):
-            return attrs
-        else:
+        """
+        get_attributes return the trainable/learnable attributes (i.e. exclude meta
```
```suggestion
        get_attributes returns the trainable/learnable attributes (i.e. exclude meta
```

jonmio commented on dataset/dataset.py:
```diff
@@ -153,56 +172,72 @@ def generate_aux_table_sql(self, aux_table, query, index_attrs=False):
             the frequency (# of entities) of a given attribute-value
-          2. self.pair_attr_stats ({ attr1 -> { attr2 -> DataFrame } } where
-            DataFrame contains 3 columns:
+          2. self.pair_attr_stats ({ attr1 -> { attr2 -> {val1 -> {val2 -> count } } } })
```
I think descriptions for a few keys in the dict are missing. 
Copied from `get_statistics` docstring
```suggestion
               3. self.pair_attr_stats ({ attr1 -> { attr2 -> {val1 -> {val2 -> count } } } })
              the statistics for each pair of attributes, attr1 and attr2, where:
                <attr1>: first attribute
                <attr2>: second attribute
                <val1>: all values of <attr1>
                <val2>: values of <attr2> that appear at least once with <val1>.
                <count>: frequency (# of entities) where attr1=val1 AND attr2=val2
        """
```

jonmio commented on dataset/dbengine.py:
```diff
@@ -98,22 +72,81 @@ def execute_query(self, query):
         return result
+
+    def _apply_func(self, func, collection):
+        if self._pool is None:
```
Why set self._pool to None when pool size is 1 instead of 1? Is it to avoid overhead incurred by Python multiprocessing library?

jonmio commented on dataset/dbengine.py:
```diff
@@ -98,22 +72,81 @@ def execute_query(self, query):
+    cur = con.cursor()
+    cur.execute(query)
+    res = cur.fetchall()
+    con.close()
```
Consider using with keyword in Python to handle db connection?

jonmio commented on dataset/table.py:
```diff
@@ -1,78 +1,88 @@
+        if self.df.empty:
+            raise Exception("Empty Dataframe associated with table {name}. Cannot return attributes.".format(
+                name=self.name))
+        return list(col for col in self.df.columns if col not in self.exclude_attr_cols)
```
Could just use list comprehension directly if a list is being returned.
```suggestion
        return [col for col in self.df.columns if col not in self.exclude_attr_cols]
```

jonmio commented on dataset/dbengine.py:
```diff
@@ -98,22 +72,81 @@ def execute_query(self, query):
+        return self._pool.map(func, collection)
+
+
+def _execute_query(args, conn_args):
```
Are docstrings needed for private fns?

jonmio commented on dcparser/dcparser.py:
```diff
@@ -4,47 +4,45 @@
     def __init__(self, env, dataset):
         """
         Constructing parser interface object
+
```
Params in docstring seem out of date

jonmio commented on domain/domain.py:
```diff
@@ -143,29 +141,34 @@ def get_active_attributes(self):
         get_corr_attributes returns attributes from self.correlations
         that are correlated with attr with magnitude at least self.cor_strength
         (init parameter).
+
```
Missing :param attr:

jonmio commented on domain/domain.py:
```diff
@@ -201,25 +207,119 @@ def generate_domain(self):
+                weak_label_idx = domain_values.index(weak_label)
+                row['weak_label'] = weak_label
+                row['weak_label_idx'] = weak_label_idx
+                row['fixed'] = CellStatus.WEAK_LABEL.value
```
At this point in the domain df, there may be rows where `weak_label` is set to some non-null value, but these cells might not be used as labelled data in the repair model because the `weak_label_prob` doesn't exceed `weak_label_threshold`. 

AFAICT this shouldn't cause any problems in the repair model because a cell is considered labelled training data if `fixed` is set to `CellStatus.WEAK_LABEL.value`. However, it might make more sense to set the weak_label for these values to null or some invalid value to avoid confusion.

jonmio commented on domain/estimator.py:
```diff
@@ -0,0 +1,48 @@
+
+    def __init__(self, dataset):
+        """
+        :param dataset: (Dataset)
```
Params in docstring out of date.

jonmio commented on domain/estimators/naive_bayes.py:
```diff
@@ -0,0 +1,66 @@
+                if at != attr:
+                    val2 = row[at]
+                    val2_count = self._freq[at][val2]
+                    val2_val1_count = 0.1
```
Could you leave a comment on what this variable is and why 0.1?

jonmio commented on domain/estimators/recurrent_logistic.py:
```diff
@@ -0,0 +1,378 @@
+    def __init__(self, dataset, pruned_domain, active_attrs):
+        """
+        :param dataset: (Dataset) original dataset
+        :param pruned_domain: (dict) of tid -> attr -> value for the domain
```
I think pruned_domain is a Dict where the key is tid (string)  and value is a Dict where the key is attr(string) and the value is a list of domain values (List[str])

jonmio commented on domain/estimators/recurrent_logistic.py:
```diff
@@ -0,0 +1,378 @@
+
+        # Featurizers used for training.
+        freq, cooccur_freq = None, None
+        if reuse_stats:
```
Has the data changed in the raw dataset? Why might we want to recompute stats here?

jonmio commented on domain/estimators/recurrent_logistic.py:
```diff
@@ -0,0 +1,378 @@
+            # later iteration with our newest current values.
+            if recur_idx > 1:
+                self._update_featurizers()
+            self._update_training_data()
```
Why is it necessary to update training data after each iteration? Does training mutate the training data?

jonmio commented on repair/featurize/constraintfeat.py:
```diff
@@ -38,28 +58,23 @@ def gen_feat_tensor(violations, total_vars, classes):
 
 
 class ConstraintFeat(Featurizer):
```
Would be nice to have a docstring for each featurizer describing either the intuition behind using these features, or what feature these featurizers are generating.

jonmio commented on repair/featurize/freqfeat.py:
```diff
@@ -5,16 +5,13 @@
-            self.single_stats[attr] = single_stats[attr].to_dict()
+        self.single_stats = single_stats
 
     def gen_feat_tensor(self, input, classes):
```
gen_feat_tensor is sometimes an instance method in some featurizers and a normal function outside of the class in others.

jonmio commented on tests/test_holoclean_repair.py:
```diff
@@ -0,0 +1,59 @@
+                    tid_col='tid',
+                    attr_col='attribute',
+                    val_col='correct_val',
+                    infer_labeled=infer_labeled)
```
Would it maybe make sense to assert that certain eval metrics like precision and recall fall within some interval so we can test that HoloClean's performance hasn't degraded significantly?

oattia committed:
Loading errors from a CSV file (#53)

* `ErrorsLoaderDetector` retrieves errors from a CSV file in the format `_tid_, attribute`

minafarid committed:
Using pytest, added tests for ErrorsLoaderDetector (#55)

richardwu committed:
Fixed get_infer_data to always return DK cells when infer_labeled =
False. Renamed parameter names for more consistency and added
estimator_enabled parameter.

minafarid committed:
Readme; conda download version

richardwu committed:
Addressed PR comments for logging stuff.

richardwu committed:
Merge pull request #57 from HoloClean/minafarid/dev-readme

Readme; conda download version

richardwu committed:
Merge pull request #56 from richardwu/dev

Fixed get_infer_data to always return DK cells when infer_labeled=False.

richardwu committed:
Fixed exception when main query fails and there is no backup query.

richardwu committed:
Fix memoization of corr attributes to also include threshold.

richardwu committed:
Return 0 for precision if total repairs is none and added logging for
weak label statistics.

richardwu committed:
Updated example with latest configuration settings.

richardwu committed:
Removed infer_labeled parameter.

richardwu committed:
Added missing DC to hospital constraints.

richardwu committed:
Added weight norm, feature norm, and (commented out) lr scheduler. Also
added LR rate to Adam.

richardwu committed:
Remove sampling prob from random domain generation.

richardwu committed:
Removed _att hospital constraints and only have one hospital constraints
file.

richardwu committed:
Removed estimator enabled parameter.

richardwu committed:
Renamed threshold parameter names.

richardwu committed:
Missing env in TiedLinear.

richardwu committed:
Renamed a bunch of featurizers and RecurrentLogistic to Logistic.

richardwu committed:
Made init feature 0/1 instead of -1/1.

richardwu committed:
Simplified Logistic to re-use training tensor for final prediction.
Reduces runtime by > 50%.

minafarid committed:
Remove NULLs from domain.

minafarid committed:
Support single predicate dcs #44, #60 (#61)

* Support 1-tuple and 2-tuple single predicate dcs #44, #60

richardwu committed:
Revert to NaiveBayes for weak labelling and revert to -1 for InitAttrFeaturizer (#63)

* Fixed naive bayes unreference thres --> self._cor_strength.
* Swap in NaiveBayes and ignore weak labels == init value.
* Move NaiveBayes prediction structures into initialization.
* Remove shuffling of training data.
* Revert InitAttr to use -1 instead of 0 for non-init domain values.
* Fix InitSim featurizer and its indexing.
* Fix test script file and set weight decay to 0.1 in example script.

minafarid committed:
Removing hospital sample constraint

minafarid committed:
Properly set random seeds (#64)

* Domain generation to correctly handle nulls
* Optimization in NB estimator
* Fix randomness in recall
* upgrade scipy version
* fix langmodel to match new gensim api

minafarid committed:
upgrade python package requirements (#65)

minafarid committed:
Correlations as normalized conditional entropy

minafarid committed:
Minor optimization in correlations

richardwu committed:
Breakout total number of repairs on ground truth into repairs on correct
and incorrect cells.

minafarid committed:
normalize conditional entropy by log base

Current performance:
```
Precision = 0.96, Recall = 0.85, Repairing Recall = 0.99,
F1 = 0.90, Repairing F1 = 0.97,
Detected Errors = 435, Total Errors = 509,
Correct Repairs = 431, Total Repairs = 450, Total Repairs (Grdth present) = 450
```

minafarid committed:
Non-learnable inits and custom feature weights

minafarid committed:
Fix super initialization for python 2

minafarid committed:
Dropping all-null columns from dataset

minafarid committed:
Fix an error in violation detection

minafarid committed:
Null detector ignore all-null columns

minafarid committed:
Reverting example

richardwu committed:
Sort domain after weak labelling.

richardwu committed:
Added logging message for number of inferred cells.

richardwu committed:
Merge pull request #70 from HoloClean/minafarid/init-featurizers

[ready] non-learnable Init featurizers

richardwu committed:
Merge pull request #68 from HoloClean/minafarid/fix-violation-detection

[ready] Fix an error in violation detection

richardwu committed:
Merge pull request #66 from HoloClean/minafarid/norm-cond-entropy-corr

[ip] Correlations as normalized conditional entropy

richardwu committed:
Merge branch 'dev' into sort_domain

richardwu committed:
Merge pull request #71 from richardwu/sort_domain

[ready] sort domain after estimator

richardwu committed:
Merge pull request #67 from richardwu/breakout_repairs_eval

[ready] Breakout total number of repairs on ground truth into repairs on correct and incorrect cells

richardwu committed:
Merge pull request #69 from HoloClean/minafarid/drop-empty-cols

Dropping all-null columns from dataset

totemw committed:
dcparser ignores comments and empty lines (#58)

richardwu committed:
Drop NULLs from ground truth and fixed lookup of values with only
NULL co-occurring values.

richardwu committed:
Remove pdb trace and uncomment example script.

richardwu committed:
Assert for precision and recall in hospital tests. (#72)

* Remove InitSim from example
and test script.
* Assert for precision and recall in tests. 
* Use abs and tolerance instead of math isclose for Python 2 compat.
* Fix integer division in test assertion for python 2 compat.

minafarid committed:
Merge branch 'master' into dev

richardwu committed:
Remove accidental comment, addressed PR comments.

richardwu committed:
Removed unnecessary parameters from predict_pp_batch.

richardwu committed:
Added helper functions for generating test database and make pytest
parallelized.

richardwu committed:
Added test for with and without InitFeaturizer.

richardwu committed:
Fix terminating unclosed connections query.

richardwu committed:
Fix wrong value for total repairs on correct groundtruth in without init
test.

richardwu committed:
Updated travis to make holocleanuser a SUPERUSER.

richardwu committed:
Use 1 worker in tests because of memory issues.

richardwu committed:
Remove LangModelFeaturizer from tests.

oattia committed:
Add support to read detected errors from a relational table. (#74)

richardwu committed:
Fix Logisitic posterior model: do not set weight decay.

richardwu committed:
Merge pull request #75 from richardwu/logistic_wdecay

Fix Logisitic posterior model: do not set weight decay.

richardwu committed:
Merge pull request #73 from richardwu/remove_nulls_from_domain

Remove NULLs from domain: always predict a non-NULL value.

richardwu committed:
Revert "Remove NULLs from domain: always predict a non-NULL value." (#77)

richardwu committed:
Remove NULLs from domain: always predict a non-NULL value.

All of the PR in one commit.

oattia committed:
Added typing information for errorloader detector and support for loading from df (#81)

oattia committed:
Adding extra example for loading errors from database (#82)
